sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -200,7 +201,6 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
     cachedDataSourceTables.getIfPresent(tableIdentifier) match {
       case null => None // Cache miss
       case logical @ LogicalRelation(relation: HadoopFsRelation, _, _) =>
-        val pathsInMetastore = metastoreRelation.catalogTable.storage.locationUri.toSeq
```

Comment:
metastoreRelation.catalogTable.storage.locationUri.toSeq

returns the base path of the relation. This is then compared to `relation.location.paths` to validate the cached entry. For non-empty partitioned tables (by that I mean partitioned tables with one or more metastore partitions), `relation.location.paths` returns the locations of the partitions. Hence, these values will never be equal and `useCached` will always be `false`.